public/app/features/manage-dashboards/components/PublicDashboardListTable.tsx

```diff
@@ -1,99 +1,115 @@
-                    fill="text"
-                    title={pd.isEnabled ? 'View public dashboard' : 'Public dashboard is disabled'}
-                    target="_blank"
-                    disabled={!pd.isEnabled}
-                  >
-                    <Icon name="external-link-alt" />
-                  </LinkButton>
-                </ButtonGroup>
-              </td>
-              <td>
-                <ButtonGroup>
-                  <LinkButton
-                    fill="text"
-                    href={`/d/${pd.dashboardUid}?shareView=share`}
-                    title="Configure public dashboard"
-                  >
-                    <Icon name="cog" />
-                  </LinkButton>
-                </ButtonGroup>
-              </td>
+    <Page.Contents isLoading={isLoading}>
+      <div className="page-action-bar">
+        <table className="filter-table">
+          <thead>
+            <tr>
+              <th className={styles.nameTh}>Name</th>
+              <th>Status</th>
+              <th className={styles.fetchingSpinner}>{isFetching && <Spinner />}</th>
             </tr>
-          ))}
-        </tbody>
-      </table>
-    </div>
+          </thead>
+          <tbody>
+            {publicDashboards?.map((pd) => (
+              <tr key={pd.uid}>
+                <td className={styles.titleTd}>
+                  <Tooltip content={pd.title} placement="top">
+                    <Link className={styles.link} href={`/d/${pd.dashboardUid}`}>
+                      {pd.title}
+                    </Link>
+                  </Tooltip>
+                </td>
+                <td>
+                  <Tag name={pd.isEnabled ? 'enabled' : 'disabled'} colorIndex={pd.isEnabled ? 20 : 15} />
+                </td>
+                <td>
+                  <ButtonGroup className={styles.buttonGroup}>
+                    <LinkButton
+                      href={viewPublicDashboardUrl(pd.accessToken)}
+                      fill="text"
+                      size={responsiveSize}
+                      title={pd.isEnabled ? 'View public dashboard' : 'Public dashboard is disabled'}
+                      target="_blank"
+                      disabled={!pd.isEnabled}
+                      data-testid={selectors.ListItem.linkButton}
+                    >
+                      <Icon size={responsiveSize} name="external-link-alt" />
+                    </LinkButton>
+                    <LinkButton
+                      fill="text"
+                      size={responsiveSize}
+                      href={`/d/${pd.dashboardUid}?shareView=share`}
+                      title="Configure public dashboard"
+                      data-testid={selectors.ListItem.configButton}
+                    >
+                      <Icon size={responsiveSize} name="cog" />
+                    </LinkButton>
+                    {hasWritePermissions && <DeletePublicDashboardButton publicDashboard={pd} size={responsiveSize} />}
```

Comment:
If we're going to break this one out, maybe we should just have a listTableComponents file with all of the buttons and import them?

Reply:
if it's a big component, it's not recommended to create it inside the same component that will use it. That's why I created a separated file. We could make a folder with all of this, though